Resolvers for item, collection and community home page

LotteHofstede opened the pull request:
This PR closes #284 
I added resolvers for the item, collection and community home page to fetch the object's data before loading the rest of the page.

This PR also contains a fix for the recent items shown on the collection page. They are now requested using the `search` endpoint instead of the `items` endpoint.

tdonohue commented on config/environment.default.js:
```diff
@@ -8,13 +8,13 @@ module.exports = {
   },
   // The REST API server settings.
-  rest: {
+    rest: {
```
This indentation is slightly odd. Looks to be accidental?  (There's another such indentation in this same config file)

tdonohue commented on src/app/+collection-page/collection-page.component.html:
```diff
@@ -35,17 +35,17 @@
       </div>
     </div>
-    <ds-error *ngIf="collectionRD?.hasFailed" message="{{'error.collection' | translate}}"></ds-error>
+    <ds-error *ngIf="collectionRD?.hasFailed"0 message="{{'error.collection' | translate}}"></ds-error>
```
There's a random `0` appended here?

tdonohue commented on src/app/+collection-page/collection-page.component.html:
```diff
@@ -35,17 +35,17 @@
           [sortConfig]="sortConfig"
           [objects]="itemRD"
-          [hideGear]="false"
+          [hideGear]="true"
```
Is there a reason to switch this from `false` to `true`?  It seems unrelated to this PR, but maybe I'm missing something?

tdonohue commented on src/app/+collection-page/collection-page.resolver.ts:
```diff
@@ -0,0 +1,20 @@
+import { getSucceededRemoteData } from '../core/shared/operators';
+
+@Injectable()
+export class CollectionPageResolver implements Resolve<RemoteData<Collection>> {
```
Please add TypeDocs to this new class

tdonohue commented on src/app/+community-page/community-page.resolver.ts:
```diff
@@ -0,0 +1,20 @@
+import { CommunityDataService } from '../core/data/community-data.service';
+
+@Injectable()
+export class CommunityPageResolver implements Resolve<RemoteData<Community>> {
```
Please add TypeDocs to this new class

tdonohue commented on src/app/+item-page/item-page.resolver.ts:
```diff
@@ -0,0 +1,19 @@
+import { Item } from '../core/shared/item.model';
+
+@Injectable()
+export class ItemPageResolver implements Resolve<RemoteData<Item>> {
```
Please add TypeDocs to this new class

LotteHofstede commented:
Thanks for your feedback, Tim
I added the missing TypeDoc for the resolvers and removed the typos.
The reason I decided to hide the gear is because I was trying to make sense of what this Recent Submissions section on a collection page (for which I changed the implementation) should actually display.
It does not make sense to give users the possibility to change the sorting configuration: Recent Submissions implies it's sorted by submission date. This feature would make sense on a browse by collection page.
